eZ/Publish/API/Repository/CompareService.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository;
+
+use eZ\Publish\API\Repository\Values\Content\VersionDiff\VersionDiff;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+
+interface CompareService
```

Comment:
Naming:
"Compare" is a verb, service names are nouns.
Also it doesn't say what it is supposed to compare. We have services for Content, Locations, Search, Bookmarks, Field Types, etc.
Maybe `ContentVersionComparisonService`? Or `ContentComparisonService`?

Though I'm not sure if moving this away from `ContentService` is such a good idea.

Reply:
Moving anything from contentService to prevent it from more bloating seems like a good idea to me.

As for the name, afaik `Compare` is also noun in some _conditions_. As you stated, we can compare a lot of things, so the question is if it would be easier to do that in one Service or across all of them.

Reply:
> As for the name, afaik `Compare` is also noun in some _conditions_.

POV ping @DominikaK 

> As you stated, we can compare a lot of things, so the question is if it would be easier to do that in one Service or across all of them.

Ok. This is not SOLID, sorry.

Reply:
What about having this as a separate service (i.e. `ContentComparisonService`) and then injected and used as a facade into `ContentService`?

Reply:
FWIW, `compare` can be a noun, but it reads like a verb and  `comparison` is much more obvious, so if you're aiming for consistent naming of services with nouns, my vote goes for `comparison`.

Reply:
ContentService is bloated now. Also, I can easily imagine that we can add method to ComparisionService which working on other objects. If we will chose `ContentComparisonService` we will end up with `OtherThingComparisonService's`. The purpose of this service is Comparing, no matter what objects. And this is SOLID I think

Reply:
> ContentService is bloated now

True.

> And this is SOLID I think when you should add to class description and compare Versions

Not sure if following, service name should reflect its purpose. While I get that you might see the single responsibility as comparing something, in the context of our domain it is not correct. Our domain responsibilities are split per well known parts as Content, Location, etc.

> So either it should be a generic 'ComparisonService' into which we inject the appropriate Comparer for each entity type, or we stick with implementing only ContentVersionComparisonService (and an interface) for now

Ok. We'll have internal sync on this today, will decide then.